docker/dev/docker-compose-publish-capybara-port.yml

```diff
@@ -0,0 +1,8 @@
+# This file will NOT be automatically included if you define COMPOSE_FILE in .env
+# You need to explicity include it in the list COMPOSE_FILE files.
+version: '3'
+services:
+  app:
+    ports:
+      - "43447:43447"
```

Comment:
Can you add a comment explaining why this is needed?  Or is there web page describing this? When running not headlessly does chrome connect to this port?

Also we'll have port conflicts with this kind of fixed port if running multiple apps with the same approach. Have you looked at ways of making this work with a random port? If the container has access to the docker socket, it can probably find the external port number, then perhaps capybara can be configured to use that port with chromedriver.

Reply:
Yes, when running features in non-headless mode the app communicates with `chromedriver` on the host machine (at port 9515). Then, on the host machine, Chrome needs to make a connection back into the Docker container at the port that the test server is running on. In this case we've chosen port `43447` but it can be anything (as long as we also configure the test server to run on that port).

I do not have experience with making this work on a random port. I'm sure it is possible but I do not see the value/need to add complexity to the setup. This would only conflict with another app if you were to choose the same port. This is not unlike running the dev server at `3000` which would also conflict with another app running at the same port at the same time.

Let me know if this makes sense, or if you'd still prefer to run this on a random port. I can investigate further.